test/new-e2e/tests/agent-metric-logs/linux-log-agent/01_file_tailing_test.go

```diff
@@ -0,0 +1,174 @@
+		checkLogs(s, "hello", "hello-before-permission-world", true)
+	}
+
+	// Part 3: Adjust permissions of new log file after log generation
+
+	// Restore permissions to default and generate log
+	output, err = s.Env().VM.ExecuteWithError("sudo chmod 644 /var/log/hello-world.log && echo true")
+	assert.NoError(t, err, "Unable to adjust back to default permissions for the log file '/var/log/hello-world.log'.")
+	if strings.TrimSpace(output) == "true" {
+		t.Logf("Permissions reset to default.")
+		generateLog(s, "hello-after-permission-world", 1)
+	}
+
+	// Grant log file permission and check intake for new logs
+	output, err = s.Env().VM.ExecuteWithError("sudo chmod +r /var/log/hello-world.log && echo true")
+	assert.NoError(t, err, "Unable to adjust permissions for the log file '/var/log/hello-world.log'.")
+	if strings.TrimSpace(output) == "true" {
+		t.Logf("Permissions granted for log file.")
+		checkLogs(s, "hello", "hello-after-permission-world", true)
+	}
+}
+
+func (s *LinuxVMFakeintakeSuite) LogPermission() {
+	t := s.T()
+	logPath := "/var/log/hello-world.log"
+	checkLogFilePresence(s, logPath)
+
+	// Part 4: Allow on only write permission to the log file so the agent cannot tail it
+	output, err := s.Env().VM.ExecuteWithError("sudo chmod -r /var/log/hello-world.log && echo true")
+	assert.NoError(t, err, "Unable to adjust permissions for the log file '/var/log/hello-world.log'.")
+	if strings.TrimSpace(output) == "true" {
+		t.Logf("Read permissions revoked")
+		s.Env().VM.Execute("sudo service datadog-agent restart")
+
+		generateLog(s, "access-denied", 1)
+		// Check intake to see if new logs are not present
+		checkLogs(s, "hello", "access-denied", false)
+	}
+
+	// Part 5: Restore permissions
+	output, err = s.Env().VM.ExecuteWithError("sudo chmod +r /var/log/hello-world.log && echo true")
+	assert.NoError(t, err, "Unable to adjust permissions for the log file '/var/log/hello-world.log'.")
+
+	// Part 6: Generate new logs and check the intake for new logs
+	if strings.TrimSpace(output) == "true" {
+		t.Logf("Permissions restored.")
+		generateLog(s, "access-granted", 1)
+		// Check intake to see if new logs are present
+		checkLogs(s, "hello", "access-granted", true)
+	}
+
+}
+
+func (s *LinuxVMFakeintakeSuite) LogRotation() {
+	t := s.T()
+	logPath := "/var/log/hello-world.log"
+	checkLogFilePresence(s, logPath)
+
+	// Part 7: Rotate the log file and check if the agent is tailing the new log file.
+	// Delete and Recreate file rotation
+	s.Env().VM.Execute("sudo mv /var/log/hello-world.log /var/log/hello-world.log.old && sudo touch /var/log/hello-world.log")
+
+	// Verify the old log file's existence after rotation
+	_, err := s.Env().VM.ExecuteWithError("ls /var/log/hello-world.log.old")
+	assert.NoError(t, err, "Failed to find the old log file after rotation")
+
+	// Grant new log file permission
+	output, err := s.Env().VM.ExecuteWithError("sudo chmod +r /var/log/hello-world.log && echo true")
```

Comment:
Since file permission isn't the goal of this test, consider utilizing `umask` to set the default file permissions for files created in `/var/log`.